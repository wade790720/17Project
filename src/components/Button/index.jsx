import React from 'react';
import PropTypes from 'prop-types';
import cx from "classnames";
import "./style/button.sass"

const Button = (props) => {
    const { className, circle, text, ...otherProps } = props;
    return (
        <button className={cx("btn", { "btn-circle": circle }, className)} {...otherProps}>
            {circle && <img src="https://img.icons8.com/android/24/000000/plus.png" />}
            {text}
        </button>
    );
};

export default Button;

Button.defaultProps = {
    circle: false,
    text: ''
}
Button.propTypes = {
    circle: PropTypes.bool,
    text: PropTypes.string
}