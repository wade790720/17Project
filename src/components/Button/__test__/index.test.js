import React from "react";
import Button from "../index";
import renderer from "react-test-renderer";

describe('Button tests', () => {
    it("Default view", () => {
        const tree = renderer.create(
            <Button />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});