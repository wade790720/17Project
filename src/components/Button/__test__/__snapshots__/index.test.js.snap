// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Default view 1`] = `
<button
  className="btn"
>
  
</button>
`;
